guides/17_Migration_to_3_0.md

```diff
@@ -162,6 +162,64 @@ handle them please remove.
 - StateMachineTransitionFailed
 - AudioDeviceSwitched
 
+## AudioVideo events
+
+The following `AudioVideo` events have been removed in v3.x, these events redirect data from `metricDidReceive`.
+
+- videoSendHealthDidChange
+- videoSendBandwidthDidChange
+- videoReceiveBandwidthDidChange
+
+In 3.x you can still replecate these events funtionality through `metricDidReceive`.
+
+
+```js
+const observer = {
+  // Before in 2.x
+  videoSendHealthDidChange: (bitrateKbps, packetsPerSecond) => {
+    console.log(`Sending video bitrate in kilobits per second: ${videoUpstreamBitrate} and ${videoUpstreamPacketPerSecond}`);
+  },
+  videoSendBandwidthDidChange: (newBandwidthKbps, oldBandwidthKbps) => {
+    console.log(`Sending bandwidth is ${availableSendBandwidth}, nack count per second is ${nackCountPerSecond}, and old bandwidth is ${this.oldSendBandwidth}`);
+  },
+  videoReceiveBandwidthDidChange: (newBandwidthKbps, oldBandwidthKbps) => {
+    console.log(`Receiving bandwidth is ${availableRecvBandwidth}, and old bandwidth is ${this.oldRecvBandwidth}`);
+  },
+  // After in 3.x
+  oldSendBandwidthKbs: 0,
+  oldRecvBandwidthKbs: 0,
+  metricsDidReceive: (clientMetricReport) => {
+    const metricReport = clientMetricReport.getObservableMetrics();
+
+    const {
+      videoPacketSentPerSecond,
+      videoUpstreamBitrate,
+      nackCountPerSecond,
+    } = metricReport;
+    const availableSendBandwidthKbs = metricReport.availableOutgoingBitrate / 1000;
+    const availableRecvBandwidthKbs = metricReport.availableIncomingBitrate / 1000;
+    // videoSendHealthDidChange
+    console.log(`Sending video bitrate in kilobits per second: ${videoUpstreamBitrate / 1000} and sending packets per second: ${videoPacketSentPerSecond}`);
+
+    // videoSendBandwidthDidChange
+    if(this.oldSendBandwidthKbs != availableSendBandwidthKbs)
+    {
+      console.log(`Sending bandwidth is ${availableSendBandwidthKbs}, nack count per second is ${nackCountPerSecond}, and old bandwidth is ${this.oldSendBandwidthKbs}`);
+      this.oldSendBandwidthKbs = availableSendBandwidthKbs;
+    }
+
+    // videoReceiveBandwidthDidChange
+    if(this.oldRecvBandwidthKbs != availableRecvBandwidthKbs)
+    {
+      console.log(`Receiving bandwidth is ${availableRecvBandwidthKbs}, and old bandwidth is ${this.oldRecvBandwidthKbs}`);
+      this.oldRecvBandwidthKbs = availableRecvBandwidthKbs;
+    }
+  },
+};
```

Comment:
Suggest splitting this into to observer.

```js
// Before in 2.x
const observer = {
  videoSendHealthDidChange: ...,
  videoSendBandwidthDidChange: ...,
  videoReceiveBandwidthDidChange: ...,
};

// After in 3,x
const observer = {
  metricsDidReceive: ...,
};
```

Reply:
Yeah that's fair, updated